### PR TITLE
IBX-425: Travis fails to report CS fixer issues in ezplatform-core

### PR DESCRIPTION
**Pull request by kacper-wieczorek-ibexa**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [IBX-425](https://issues.ibexa.co/browse/IBX-425)
| **Bug/Improvement**| yes/no
| **New feature**    | yes/no
| **Target version** | 
| **BC breaks**      | yes/no
| **Tests pass**     | yes/no
| **Doc needed**     | yes/no

**TODO**:
- [ ] Implement feature / fix a bug.
- [ ] Implement tests.
- [ ] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [ ] Ask for Code Review.

### CODE REVIEW

**Comment by Steveb-p** [comment]
Use github action instead.

See https://github.com/ibexa/migrations/blob/master/.github/workflows/ci.yml

File: composer.json [hunk 1]
```diff
@@ -30,7 +30,8 @@
     },
     "scripts": {
         "test": "phpunit -v -c phpunit.xml",
-        "fix-cs": "@php ./vendor/bin/php-cs-fixer fix -v --show-progress=estimating"
+        "fix-cs": "@php ./vendor/bin/php-cs-fixer fix -v --show-progress=estimating",
```
Comment (adamwojs) [inline composer.json]:
```suggestion
        "fix-cs": "php-cs-fixer fix -v --show-progress=estimating",
```

File: composer.json [hunk 2]
```diff
@@ -30,7 +30,8 @@
...
+        "run-cs": "@php ./vendor/bin/php-cs-fixer fix -v --dry-run --show-progress=estimating"
```
Comment (adamwojs) [inline composer.json]:
```suggestion
        "run-cs": "php-cs-fixer fix -v --dry-run --show-progress=estimating"
```

**Review by adamwojs — COMMENTED** [review]
Code style should be aligned with https://github.com/ezsystems/ezplatform-code-style

File: composer.json (hunk 2 above)
Comment (Steveb-p) [reply composer.json]: Some packages use `check-cs`, which version should we stick to to prevent mismatch issues?

File: composer.json [hunk 3]
```diff
@@ -26,11 +26,13 @@
     },
     "require-dev": {
         "friendsofphp/php-cs-fixer": "^2.15",
-        "phpunit/phpunit": "^8.3.5"
+        "phpunit/phpunit": "^8.3.5",
+        "ezsystems/ezplatform-code-style": "^1.0@dev"
```
Comment (Steveb-p) [inline composer.json]: Keep the ordering alphabetical. Ideally we'd have `config.sort-packages: true` defined, but I'm not sure if `require` packages are already ordered or not.

Comment (kacper-wieczorek-ibexa) [reply composer.json]: I used function, You suggested. Check it out now. :)

**Review by Steveb-p — APPROVED** [review]
:+1:, although I'm pretty sure that those exception changes will need to be reverted as they break soft cap on line length (which isn't enforced by code style. Yet.)

File: composer.json (hunk 2 above)
Comment (mnocon) [reply composer.json]: Looks like `check-cs` is more popular than `php-cs`:
Zrzut ekranu 2021-06-18 o 10 49 24

Zrzut ekranu 2021-06-18 o 10 48 26

If you're ok with it I'd go with `check-cs` here as well (and maybe, as a follow-up, change the `php-cs` occurrences into `check-cs` to keep things consistent across packages).

File: composer.json [hunk 4]
```diff
@@ -25,16 +25,21 @@
...
     "require-dev": {
+        "ezsystems/ezplatform-code-style": "^1.0@dev",
         "friendsofphp/php-cs-fixer": "^2.15",
         "phpunit/phpunit": "^8.3.5"
     },
     "scripts": {
         "test": "phpunit -v -c phpunit.xml",
-        "fix-cs": "@php ./vendor/bin/php-cs-fixer fix -v --show-progress=estimating"
+        "fix-cs": "php-cs-fixer fix -v --show-progress=estimating",
+        "php-cs": "php-cs-fixer fix -v --diff --dry-run --show-progress=estimating"
```
Comment (mnocon) [inline composer.json]:
```suggestion
        "check-cs": "php-cs-fixer fix -v --diff --dry-run --show-progress=estimating"
```

As mentioned in the https://github.com/ezsystems/ezplatform-core/pull/39#discussion_r654259967 , for consistency with other packages

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      - uses: "ramsey/composer-install@v1"
+        with:
+          dependency-versions: "highest"
+          composer-options: "--prefer-dist --no-progress"
+
+      - name: Setup problem matchers for PHPUnit
+        run: echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+
+      - name: Run code style check
+        run: composer run-script php-cs -- --format=checkstyle | cs2pr
```
Comment (mnocon) [inline .github/workflows/ci.yaml]:
```suggestion
        run: composer run-script check-cs -- --format=checkstyle | cs2pr
```

**Review by micszo — APPROVED** [review]

**Review by mnocon — APPROVED** [review]

**Pull request merged**